File: anistats_cli/src/commands/serve.rs
use anistats_common::config::{database::DatabaseConfig, server::ServerConfig};

pub async fn serve(server_config: ServerConfig, database_config: DatabaseConfig) -> anyhow::Result<()> {
    anistats_core::serve(
        server_config,
        database_config,
    ).await?;

    Ok(())
}

File: internal/grpc/serve.go
package grpc

import (
	"fmt"
	"log/slog"
	"net"

	"gitlab.jojoxd.nl/jojoxd/anistats/internal/config"
	"gitlab.jojoxd.nl/jojoxd/anistats/internal/dbal"
)

func Serve(cfg *config.Config) error {
	db, err := dbal.New(cfg.Database)
	if err != nil {
		return err
	}

	defer db.Close()

	server := New(db)

	addr := fmt.Sprintf("%s:%d", cfg.Server.Host, cfg.Server.Port)
	listener, err := net.Listen("tcp", addr)
	if err != nil {
		return err
	}

	slog.Info("Serving", "Address", addr)

	err = server.Serve(listener)

	return err
}
